doc/quickstart/configure.rst

```diff
@@ -161,6 +180,199 @@ the user.
    ``--argument_name argument_value``
 
 
+.. _config-esgf:
+
+ESGF configuration
+==================
+
+The ``esmvaltool run`` command can automatically download the files required
+to run a recipe from ESGF for the projects CMIP3, CMIP5, CMIP6, CORDEX, and obs4MIPs.
+The downloaded files will be stored in the ``download_dir`` specified in the
+:ref:`user configuration file`.
+To enable automatic downloads from ESGF, set ``offline: false`` in
+the :ref:`user configuration file` or provide the command line argument
+``--offline=False`` when running the recipe.
+
+.. note::
+
+   When running a recipe that uses many or large datasets on a machine that
+   does not have any data available locally, the amount of data that will be
+   downloaded can be in the range of a few hundred gigabyte to a few terrabyte.
+   See :ref:`esmvaltool:inputdata` for advice on getting access to machines
+   with large datasets already available.
+
+   A log message will be displayed with the total amount of data that will
+   be downloaded before starting the download.
+   If you see that this is more than you would like to download, stop the
+   tool by pressing the ``Ctrl`` and ``C`` keys on your keyboard simultaneously
+   several times, edit the recipe so it contains fewer datasets and try again.
+
+For downloading some files (e.g. those produced by the CORDEX project),
+you need to log in to be able to download the data.
+
+See the
+`ESGF user guide <https://esgf.github.io/esgf-user-support/user_guide.html>`_
+for instructions on how to create an ESGF OpenID account if you do not have
+one yet.
+Note that the OpenID account consists of 3 components instead of the usual
+two, in addition a username and password you also need the hostname of the
+provider of the ID; for example
+`esgf-data.dkrz.de <https://esgf-data.dkrz.de/user/add/?next=http://esgf-data.dkrz.de/projects/esgf-dkrz/>`_.
+Even though the account is issued by a particular host, the same OpenID
+account can be used to download data from all hosts in the ESGF.
+
+Next, configure your system so the ``esmvaltool`` can use your credentials.
+This can be done using the keyring_ package or they can be stored in a
+:ref:`configuration file <config_esgf_pyclient>`.
+
+.. _keyring:
+
+Storing credentials in keyring
+------------------------------
+First install the keyring package. Note that this requires a supported
+backend that may not be available on compute clusters, see the
+`keyring documentation <https://pypi.org/project/keyring>`__ for more
+information.
+
+.. code-block:: bash
+
+    pip install keyring
+
+Next, set your username and password by running the commands:
+
+.. code-block:: bash
+
+    keyring set ESGF hostname
+    keyring set ESGF username
+    keyring set ESGF password
+
+for example
+
+.. code-block:: bash
+
+    keyring set ESGF esgf-data.dkrz.de
```

Comment:
This should be like this, no?

```suggestion
    keyring set ESGF hostname esgf-data.dkrz.de
```

Reply:
Yes, thanks, I'll change it!

Reply:
I tried it, but I got
```
$ keyring set ESGF hostname esgf-data.dkrz.de
usage: keyring [-h] [-p KEYRING_PATH] [-b KEYRING_BACKEND] [--list-backends] [--disable] [operation] [service] [username]
keyring: error: unrecognized arguments: esgf-data.dkrz.de
```

Any idea what went wrong?

Reply:
Could it be related to the current shutdown at DKRZ?

> The shutdown will start on Monday September 20th at 06:00am and we expect Mistral back on Wednesday 22nd in the afternoon. During the downtime no login is possible to9 Mistral, Eddy, ClimXtreme (miklip). Also other services like CERA and ESGF are affected by this downtime.

Reply:
No. This is a purely local change in configuration. I don't even think that keyring considers whether this is a hostname.

Reply:
Indeed, you need to run `keyring set ESGF hostname` and then enter the hostname as the password. I'll update the documentation.

Reply:
Sorry for the confusion, I fixed the example text. Fingers crossed that it works!

Reply:
Seems to work. Thanks!